main.go

```diff
@@ -269,148 +374,62 @@ func keygenCmd(c *cli.Context) error {
+	if !c.Args().Present() || (c.NArg() < 3) && !c.IsSet("group") {
+		slog.Fatal("group command take at least 3 keys as arguments")
 	}
 	var threshold = key.DefaultThreshold(c.NArg())
-	if c.IsSet("threshold") {
-		if c.Int("threshold") < threshold {
-			slog.Print("WARNING: You are using a threshold which is TOO LOW.")
-			slog.Print("		 It should be at least ", threshold)
-		}
-		threshold = c.Int("threshold")
-	}
-
 	publics := make([]*key.Identity, c.NArg())
-	for i, str := range args {
+	for i, str := range c.Args() {
 		pub := &key.Identity{}
 		slog.Print("Reading public identity from ", str)
 		if err := key.Load(str, pub); err != nil {
 			slog.Fatal(err)
 		}
 		publics[i] = pub
 	}
-	group := key.NewGroup(publics, threshold)
-	groupPath := path.Join(fs.Pwd(), gname)
-	if c.String("out") != "" {
-		groupPath = c.String("out")
-	}
-	if err := key.Save(groupPath, group, false); err != nil {
-		slog.Fatal(err)
-	}
-	slog.Printf("Group file written in %s. Distribute it to all the participants to start the DKG", groupPath)
-	return nil
-}
-
-func dkgCmd(c *cli.Context) error {
-	if !c.IsSet("group-init") {
-		slog.Fatal("dkg requires a group.toml file")
-	}
-	group := getGroup(c)
-	conf := contextToConfig(c)
-	if exit := resetBeaconDB(conf); exit {
-		os.Exit(0)
-	}
-	fs := key.NewFileStore(conf.ConfigFolder())
-	drand, err := core.NewDrand(fs, group, conf)
-	if err != nil {
-		slog.Fatal(err)
-	}
-	return runDkg(c, drand, fs)
-}
-
-func runDkg(c *cli.Context, d *core.Drand, ks key.Store) error {
-	var err error
-	if c.Bool("leader") {
-		err = d.StartDKG()
-	} else {
-		err = d.WaitDKG()
-	}
-	if err != nil {
-		slog.Fatal(err)
-	}
-	slog.Print("DKG setup finished!")
-
-	public, err := ks.LoadDistPublic()
-	if err != nil {
-		slog.Fatal(err)
-	}
-	dir := fs.Pwd()
-	p := path.Join(dir, dpublic)
-	key.Save(p, public, false)
-	slog.Print("distributed public key saved at ", p)
-	return nil
-}
-
-func beaconCmd(c *cli.Context) error {
-	conf := contextToConfig(c)
-	fs := key.NewFileStore(conf.ConfigFolder())
-	drand, err := core.LoadDrand(fs, conf)
-	if err != nil {
-		slog.Fatal(err)
-	}
-	drand.BeaconLoop()
-	return nil
-}
-
-func runCmd(c *cli.Context) error {
-	//x509 not available on windows: must run in insecure mode
-	if runtime.GOOS == "windows" && !c.Bool("insecure") {
-		slog.Fatal("TLS is not available on Windows, please run in insecure mode")
-	}
-	conf := contextToConfig(c)
-	fs := key.NewFileStore(conf.ConfigFolder())
-	var drand *core.Drand
-	var err error
-	if c.IsSet("group-init") {
-		group := getGroup(c)
-		if exit := resetBeaconDB(conf); exit {
-			os.Exit(0)
+	if c.IsSet("group") {
+		groupPath := c.String("group")
+		oldG := &key.Group{}
+		if err := key.Load(groupPath, oldG); err != nil {
+			slog.Fatal(err)
 		}
-		drand, err = core.NewDrand(fs, group, conf)
-		if err != nil {
+		group := oldG.MergeGroup(publics)
+		if err := key.Save(groupPath, group, false); err != nil {
```

Comment:
We should not save the new group toml file at this place since this is not YET the current group.toml file, but only the next one in place. Only when the resharing method has been called and finished that we can save the new group.toml file at this place. I'll take care of it in my PR.